packages/ui-extensions/src/types/Heading.ts

```diff
@@ -0,0 +1,20 @@
+type Level = 1 | 2 | 3;
+
+export interface BaseHeadingProps {
+  /**
+   * Unique identifier. Typically used to make the heading a target
+   * that another component can refer to in order to provide an alternative
+   * accessibility label.
+   */
+  id?: string;
+  /**
+   * The visual level of the heading.
+   */
+  level?: Level;
+  /**
+   * A custom accessibility role for this heading. Currently, only `presentation`
+   * is supported. `presentation` will strip the semantic meaning of the heading,
+   * but leave the visual styling intact.
+   */
+  role?: 'presentation';
```

Comment:
Should this be an enum?

Reply:
I think this could be an enum, but we use this string pattern everywhere ([Button](https://github.com/Shopify/argo/blob/8a4948ad1f2bdd4a71c9e5da331de57756f19541/packages/ui-extensions/src/types/Button.ts), [Banner](https://github.com/Shopify/argo/blob/8a4948ad1f2bdd4a71c9e5da331de57756f19541/packages/ui-extensions/src/types/Banner.ts), etc.). We are restricting the type to this exact string so it's pretty safe.